src/wexond/extensions/components.ts

```diff
@@ -0,0 +1,22 @@
+import { components } from 'electron'
+import { getLogger } from '../../frontend/src/logging/loggers'
+
+const log = getLogger('Components')
+
+let initializing: Promise<void[]> | undefined
+
+export default async function initComponents(): Promise<void> {
+  // eslint-disable-next-line @typescript-eslint/no-misused-promises
+  if (initializing) {
+    await initializing
```

Comment:
I'm confused with this line. If I understand correctly, initializing is initially undefined. It will get defined at line 16 (await (initializing = components.whenReady())). This promise resolve void which sounds like it could not be waited for. And therefore, why should we wait for it to resolve ?

Unless, component.whenReady() returns a promise which has to be executed by await initializing ?

Reply:
We do not care for the result, we only wait for Electron to finish loading the components which is when the Promise resolves
The only reason for the `initializing` variable is that any subsequent calls to `initComponents` made when the first one is still pending should not trigger another initialization from Electron. This is a safeguard because I do not know the internal implementation of this function.

Actually it would have been perfectly safe not to reset initializing to undefined if the first `components.whenReady()` succeeded, but it seems that Electron is already doing something similar

Reply:
Sadly it's not documented on Electron's website, but here is the (partial) signature for "components"
<img width="518" alt="image" src="https://user-images.githubusercontent.com/5877409/203502678-339f3b02-1f6c-4615-9ea5-9e1f0232eae3.png">